rust/connlib/libs/common/src/messages.rs

```diff
@@ -26,7 +26,7 @@ pub struct Peer {
     /// Peer's Ipv6 (only 1 ipv6 per peer for now and mandatory).
     pub ipv6: Ipv6Addr,
     /// Preshared key for the given peer.
-    pub preshared_key: Key,
+    pub preshared_key: SecretKey,
```

Comment:
@conectado Spent a bit more time on this tonight, I think I'll need a quick pairing session tomorrow to resolve this -- unfortunately the compiler is complaining that PartialEq, Eq, and Debug are not satisfied here.

`impl DebugSecret for Key {}` solves the Debug issue (but allows the SecretKey to be printed), and since `Secret` is defined outside this crate, I'm not able to `impl PartialEq for SecretKey` here.

Reply:
It may be the case that `Key` and `SecretKey` are different modules with their own versions of the derived `Peer` traits to get around this issue.